Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -584,40 +588,41 @@ frida_code = """
 
 process = frida.get_usb_device().attach("Safari")
 script = process.create_script(frida_code)
-script.on('message', message_callback)
 script.load()
 
 sys.stdin.read()
 ```
 
-Start Safari on the iOS device. Run the above Python script on your connected host and open the device log (we'll explain how to open device logs in the following section). Try opening a new URL in Safari; you should see Frida's output in the logs.
+Start Safari on the iOS device. Run the above Python script on your connected host and open the device log (as explained in the section "Monitoring System Logs" from the chapter "iOS Basic Security Testing"). Try opening a new URL in Safari, e.g. <https://github.com/OWASP/owasp-mstg>; you should see Frida's output in the logs as well as in your terminal.
 
-![Frida Xcode Log](Images/Chapters/0x06b/frida-xcode-log.jpg)
+![Frida Xcode Log](Images/Chapters/0x06c/frida-xcode-log.png)
 
-Of course, this example illustrates only one of the things you can do with Frida. To unlock the tool's full potential, you should learn to use its JavaScript API. The documentation section of the Frida website has a [tutorial](https://www.frida.re/docs/ios/ "Frida iOS Tutorial") and [examples](https://www.frida.re/docs/examples/ios/ "Frida iOS examples") of Frida usage on iOS.
+Of course, this example illustrates only one of the things you can do with Frida. To unlock the tool's full potential, you should learn to use its [JavaScript API](https://www.frida.re/docs/javascript-api/ "Frida JavaScript API reference"). The documentation section of the Frida website has a [tutorial](https://www.frida.re/docs/ios/ "Frida Tutorial") and [examples](https://www.frida.re/docs/examples/ios/ "Frida examples") for using Frida on iOS.
 
-Please also take a look at the [Frida JavaScript API reference](https://www.frida.re/docs/javascript-api/).
+###### Cycript
 
-### Patching React Native Applications
-
-If the [React Native](https://facebook.github.io/react-native "React Native") framework has been used for development, the main application code is in the file `Payload/[APP].app/main.jsbundle`. This file contains the JavaScript code. Most of the time, the JavaScript code in this file is minified. With the tool [JStillery](https://mindedsecurity.github.io/jstillery "JStillery"), a human-readable version of the file can be retried, which will allow code analysis. The [CLI version of JStillery](https://github.com/mindedsecurity/jstillery/ "CLI version of JStillery") and the local server are preferable to the online version because the latter discloses the source code to a third party.
+- Install the application that will be hooked.
```

Comment:
it is a bit odd to have enumeration directly here...

Reply:
Actually I agree. That was already there but I can fix it by putting a sentence before.

Reply:
actually this is not really method hooking, but it is titled like that in master "Hooking Native Functions and Objective-C Methods". It is simply basic Cycript usage.

I'll merge it with Cycript's intro in "Dynamic Instrumentation -> Tooling". If someone wants to write about method hooking with Cycript we can open an issue (?)

Reply:
I've done that and referenced Cycript Manual - <http://www.cycript.org/manual/>

I've removed the printMethods because it actually does not exist (is not built-in) and replaced it with a built-in function: choose(). The rest is basically the same content as before, just a bit more organized now.